pumpkin-lib/src/engine/nogoods/literal_block_distance.rs

```diff
@@ -0,0 +1,45 @@
+use crate::containers::SparseSet;
+use crate::engine::Assignments;
+use crate::predicates::Predicate;
+
+/// Used to compute the LBD of nogoods. The type carries state that prevents the re-allocation of
+/// helper data structures.
+#[derive(Clone, Debug)]
+pub(crate) struct Lbd {
```

Comment:
what do you think about changing the name to `LbdComputer` or `LbdCalculator`? LBD is normally just a single number so this way we could distinguish between the struct that computes the LBD and the actual LBD value.

Another point: we could also expand name LBD into LiteralBlockDistance[Calculator] for the struct name, what do you think?